### PR TITLE
Update context action objects

### PR DESCRIPTION
**Pull request by guscost**

### CODE REVIEW

File: src/contextualizers/heuristics/erc20Mint/erc20Mint.ts [hunk 1]
```diff
@@ -117,7 +118,8 @@ export function generate(transaction: Transaction): Transaction {
       },
       minted: {
         type: 'contextAction',
-        value: HeuristicContextActionEnum.MINTED,
+        id: HeuristicContextActionEnum.MINTED,
+        value: HeuristicContextActionEnum.MINTED.split(".").pop() || "",
```
Comment (guscost) [inline src/contextualizers/heuristics/erc20Mint/erc20Mint.ts:122]: This does nothing for heuristic (non-prefixed) actions, but is left here as an example of how the value would be calculated.

Comment (pcowgill) [reply src/contextualizers/heuristics/erc20Mint/erc20Mint.ts:122]: I was assuming that as a rule no enums would contain a . If we ever want the string containing a ".", we compute that from the enum and the protocol name

Comment (pcowgill) [reply src/contextualizers/heuristics/erc20Mint/erc20Mint.ts:122]: That would mean that for id and for `actions` above, we add the "." and the protocol name

Comment (pcowgill) [reply src/contextualizers/heuristics/erc20Mint/erc20Mint.ts:122]: Maybe we can make the change in a protocol contextualizer to show how it behaves in the case where there is a protocol name too

Comment (guscost) [reply src/contextualizers/heuristics/erc20Mint/erc20Mint.ts:122]: I guess that makes sense, fewer type changes too. OK will try that way, 1sec

File: src/contextualizers/protocol/basepaint/basepaint.ts [hunk 2]
```diff
@@ -64,6 +65,7 @@ export const generate = (transaction: Transaction): Transaction => {
         variables: {
           contextAction: {
             type: 'contextAction',
+            id: `${Protocols.BASEPAINT}.${BasepaintActionEnum.PAINTED}`,
```
Comment (pcowgill) [inline src/contextualizers/protocol/basepaint/basepaint.ts:68]: @guscost I think `summaries.en.title` should be `Protocols.BASEPAINT` too

File: src/types/context.ts [hunk 3]
```diff
@@ -30,6 +36,10 @@ export type ContextCodeType = {
 
 export type ContextActionType = {
   type: 'contextAction';
+  id:
```
Comment (pcowgill) [inline src/types/context.ts:39]: Nice!

**Review by pcowgill — CHANGES_REQUESTED** [review]
We'll need to add the `actions` array at the same level as variables and summaries for each contextualization in this PR too

File: src/contextualizers/heuristics/erc20Mint/erc20Mint.ts [hunk 4]
```diff
@@ -100,6 +100,7 @@ export function generate(transaction: Transaction): Transaction {
       : '0';
 
   transaction.context = {
+    actions: [HeuristicContextActionEnum.MINTED],
```
Comment (pcowgill) [inline src/contextualizers/heuristics/erc20Mint/erc20Mint.ts:103]: We have actions here, so the remaining work is to propagate this change to other contextualizers

File: src/contextualizers/protocol/basepaint/basepaint.ts [hunk 5]
```diff
@@ -88,7 +91,7 @@ export const generate = (transaction: Transaction): Transaction => {
         summaries: {
           category: 'PROTOCOL_1',
           en: {
-            title: 'Basepaint',
+            title: ProtocolMap[Protocols.BASEPAINT],
```
Comment (pcowgill) [inline src/contextualizers/protocol/basepaint/basepaint.ts:94]: This approach is nice

File: src/types/contextAction/index.ts [hunk 6]
```diff
@@ -1,12 +1,17 @@
...
-import { ProtocolContextAction } from './protocolContextAction';
+import { ProtocolContextAction, Protocols } from './protocolContextAction';
 import { BridgeContextAction } from './bridge';
 
 export type ContextAction =
   | HeuristicContextAction
   | ProtocolContextAction
   | BridgeContextAction;
 
+export type ContextActionFull =
```
Comment (pcowgill) [inline src/types/contextAction/index.ts:10]: Maybe `ContextActionID` would be a more consistent name with the way we use id within variables of type `contextAction`
```suggestion
export type ContextActionID =
```

**Review by pcowgill — COMMENTED** [review]
Other than the requested changes, I reviewed the rest of the PR, and it looks great!

File: src/contextualizers/heuristics/erc20Mint/erc20Mint.ts (hunk 4 above)
Comment (guscost) [reply src/contextualizers/heuristics/erc20Mint/erc20Mint.ts:103]: Done!

File: src/contextualizers/bridges/acrossProtocol/source.ts [hunk 7]
```diff
@@ -75,13 +75,16 @@ export function generate(transaction: Transaction): Transaction {
   );
   const amount = formatEther(fundsDepositedEvent.args['amount'] as bigint);
   transaction.context = {
+    actions: [HeuristicContextActionEnum.BRIDGED],
+
```
Comment (pcowgill) [inline src/contextualizers/bridges/acrossProtocol/source.ts:79]: @guscost This newline isn't ideal, but doesn't matter

**Pull request merged**